lib/pengine/rules.c

```diff
@@ -569,39 +569,33 @@ accept_attr_expr(const char *l_val, const char *r_val, enum pcmk__type type,
  * \internal
  * \brief Get correct value according to \c PCMK_XA_VALUE_SOURCE
  *
- * \param[in] expr_id       Rule expression ID (for logging only)
- * \param[in] value         value given in rule expression
- * \param[in] value_source  \c PCMK_XA_VALUE_SOURCE given in rule expressions
- * \param[in] match_data    If not NULL, resource back-references and params
+ * \param[in] value       value given in rule expression
+ * \param[in] source      Reference value source
+ * \param[in] match_data  If not NULL, resource back-references and params
```

Comment:
There's only one caller, and it guarantees `match_data` is non-NULL. If it *were* NULL, then `source` would have to be `pcmk__source_literal`, or we'd segfault.

Reply:
This gets overhauled in next pr